File: AOC2020/day01/day01.js
var fs=require("fs");
let input = fs.readFileSync("./input.txt" ).toString().split("\n").map(x=>parseInt(x));
input.filter(x=> input.find(y => y==2020-x)!=undefined).reduce((prod,acc)=> prod*acc, 1);
input.filter(x=> input.find(y => y==2020-x-(input.filter(z => z==2020-x-y)[0]))!=undefined).reduce((prod,acc)=> prod*acc, 1);

File: AOC2020/day01/day01.py
# %%
with open('input.txt') as txt: s=txt.read() 
l = [int(i) for i in s.split()]

# %%
from functools import reduce
reduce(lambda x,y : x*y,[i for i in l if 2020-i in l])
# %%
from itertools import product
set([x*y*z for x,y,z in product(l,l,l) if x+y+z==2020])

#%%